include/libtorrent/kademlia/dht_state.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+
+Copyright (c) 2016, Arvid Norberg, Alden Torres
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in
+      the documentation and/or other materials provided with the distribution.
+    * Neither the name of the author nor the names of its
+      contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef LIBTORRENT_DHT_STATE_HPP
+#define LIBTORRENT_DHT_STATE_HPP
+
+#include <libtorrent/config.hpp>
+#include <libtorrent/socket.hpp>
+#include <libtorrent/entry.hpp>
+
+#include <libtorrent/kademlia/node_id.hpp>
+
+#include <vector>
+
+namespace libtorrent
+{
+	struct bdecode_node;
+}
+
+namespace libtorrent {
+namespace dht
+{
+	// This structure helps to store and load the state
+	// of the ``dht_tracker``.
+	// At this moment the library is only a dual stack
+	// implementation of the DHT. See BEP38_
+	//
+	// .. _BEP32: http://bittorrent.org/beps/bep_0032.html
+	struct TORRENT_EXPORT dht_state
+	{
+		dht_state() = default;
+		dht_state(dht_state const&) = default;
+		dht_state(dht_state&&) = default;
+		dht_state& operator=(dht_state const&) = default;
+		dht_state& operator=(dht_state&&) = default;
```

Comment:
I'm pretty sure if you don't specify any of these, you get the same behavior. (see [rule of zero](https://rmf.io/cxx11/rule-of-zero/))